book-3-workforce/chapters/ASPNET_INTRO.md

```diff
@@ -84,19 +84,41 @@ public IDbConnection Connection
 Your `Index` method in a web application should return a list of the corresponding resource. Modify your method to contain the following code.
 
 ```cs
-using (IDbConnection conn = Connection)
+using (SqlConnection conn = Connection)
 {
-    var students = await conn.QueryAsync<Student>(@"
-        SELECT s.Id
-            s.FirstName,
-            s.LastName,
-            s.SlackHandle,
-            s.CohortId
-        FROM Student s
-    ");
-    return View(students);
+    conn.Open();
+    using (SqlCommand cmd = conn.CreateCommand())
+    {
+        cmd.CommandText = @"
+            SELECT s.Id
+                s.FirstName,
+                s.LastName,
+                s.SlackHandle,
+                s.CohortId
+            FROM Student s
+        ";
+        SqlDataReader reader = cmd.ExecuteReader();
+
+        List<Student> students = new List<Student>();
+        while (reader.Read())
+        {
+            Student student = new Student
+            {
+                Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                FirstName = reader.GetString(reader.GetOrdinal("FirstName")),
+                LastName = reader.GetString(reader.GetOrdinal("LastName")),
+                SlackHandle = reader.GetString(reader.GetOrdinal("SlackHandle")),
+                CohortId = reader.GetInt32(reader.GetOrdinal("CohortId"))
+            };
+
+            students.Add(student);
+        }
+
+        reader.Close();
+
+        return View(students);
+    }
 }
-
 ```
 
 ## Student List Razor Template
```

Comment:
The gif on line 132 shows code that uses Dapper.

Reply:
Removed